Userland/Applications/PixelPaint/FilterTool.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021, Tobias Christiansen <tobyase@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "Tool.h"
+#include <AK/OwnPtr.h>
+#include <AK/Vector.h>
+#include <LibGfx/Filters/Filter.h>
+
+namespace PixelPaint {
+
+class FilterTool final : public Tool {
+public:
+    FilterTool();
+
+    virtual GUI::Widget* get_properties_widget() override;
+
+private:
+    RefPtr<GUI::Widget> m_properties_widget;
+
+    enum class Filters {
+        Unselected,
+        LaplacianCardinal,
+        LaplacianDiagonal,
+        Gauss3,
+        Gauss5,
+        BoxBlur3,
+        BoxBlur5,
+        Sharpen,
+        Grayscale,
+        Invert,
+        __Count,
+    };
+
+    Filters m_selected_filter { Filters::Unselected };
```

Comment:
`Filters` with an `s` seems like a bit of a weird name for an enum. Maybe `FilterType`?

Reply:
Also, maybe just have `Optional<FilterType>` instead of having "Unselected" be a part of the enum?

Reply:
It's easiest to just keep the ::Unselected imo, since I loop over the whole enum to initialize the vectors and "Select Filter..." is a valid selection.